### DISCUSSION TITLE
.astro file templating as standalone project

### DISCUSSION THREAD

**Discussion by cdtut** [withastro/roadmap#768] [Proposal]
### Body

Templating from astro is nice but we are already using other server side frameworks. We would like to use astro templating instead of other templating engines like nunjucks or EJS. Can templating be made into standalone project?

Then we can render the files like other templating engines `astro.render("template.astro", data))`. Data parameter would be merged with what is in front matter which can be optional.

**Comment by mariohamann** [comment]
What you're describing should be doable with the proposed Container API (being already in stage 2): https://github.com/withastro/roadmap/issues/533
